### PR TITLE
Use sbt-release to do releases

### PR DESCRIPTION
**Pull request by marcospereira**
Instead of using a script which only runs `+publishSigned` use
the full release process from sbt-release, including sonatype
promotion of artifacts.

Follow up to #95.

Depends on #103.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -1,67 +1,26 @@
...
+echo "Starting release. You will be prompt to confirm the version and push changes at the end of the process"
+sbt 'release cross'
 
-# fail the script with an error message
-function fail {
-  echoerr "$@"
-  exit 1
-}
+# Get the latest tag which is the one created above
+version=$(git tag --sort=-taggerdate | head -1)
```
Comment (marcospereira) [inline release.sh]: Only for "debug" purposes.

File: release.sh [hunk 2]
```diff
@@ -1,67 +1,26 @@
...
+#!/usr/bin/env bash
 
-# print usage info
-function usage {
-  cat <<EOM
-Usage: ${script_name} VERSION
-EOM
-}
+echo "Starting release. You will be prompt to confirm the version and push changes at the end of the process"
+sbt 'release cross'
```
Comment (marcospereira) [inline release.sh]: sbt-release will take care of most of what was done by this script.

File: release.sh [hunk 3]
```diff
@@ -1,67 +1,26 @@
...
-git commit version.sbt -m "Update version to $version"
-
-echo "Publishing artifacts..."
-sbt '+publishSigned'
-
-echo "Publishing docs..."
+echo "Publishing documentation for version $version"
 sbt makeSite
-mv documentation/target/site ../releasing-the-docs
+mv -v documentation/target/site ../releasing-the-docs
```
Comment (marcospereira) [inline release.sh]: Making some commands `-v`.

File: release.sh [hunk 4]
```diff
@@ -1,67 +1,26 @@
...
+git commit -m "Releasing docs for version $version"
 git push origin gh-pages
 
-echo "Tagging release..."
-git tag -m "Releasing $version" "v$version"
-git push --tags
-
-git checkout master
+echo "Docs where published. Getting back to release branch."
+git cd -
```
Comment (marcospereira) [inline release.sh]: See #102.

File: release.sh [hunk 5]
```diff
@@ -1,67 +1,31 @@
...
+git commit -m "Releasing docs for version $version"
 git push origin gh-pages
 
-echo "Tagging release..."
-git tag -m "Releasing $version" "v$version"
-git push --tags
-
-git checkout master
+echo "Docs where published. Getting back to release branch."
+git checkout -
```
Comment (marcospereira) [inline release.sh]: See #102.

**Comment by dwijnand** [comment]
If no one has strong feelings against this, let's land these changes!

**Comment by dwijnand** [comment]
Bueno

**Pull request merged**